code/datums/components/proximity_monitor.dm

```diff
@@ -1,139 +1,472 @@
+	hasprox_receiver = null
+	host = null
 	QDEL_LIST(proximity_checkers)
-	owner = null
 	return ..()
 
 /datum/component/proximity_monitor/RegisterWithParent()
 	. = ..()
-	if(ismovable(parent))
-		RegisterSignal(parent, COMSIG_MOVABLE_MOVED, .proc/HandleMove)
+	// Override vars are set to TRUE here to allow for component transfering.
+	if(ismovable(hasprox_receiver))
+		RegisterSignal(hasprox_receiver, COMSIG_MOVABLE_MOVED, .proc/handle_move, TRUE)
+	if(host && ismovable(host))
+		RegisterSignal(host, COMSIG_MOVABLE_MOVED, .proc/handle_move, TRUE)
 
 /datum/component/proximity_monitor/UnregisterFromParent()
 	. = ..()
-	if(ismovable(parent))
-		UnregisterSignal(parent, COMSIG_MOVABLE_MOVED)
+	if(ismovable(hasprox_receiver))
+		UnregisterSignal(hasprox_receiver, COMSIG_MOVABLE_MOVED)
+	if(host && ismovable(host))
+		UnregisterSignal(host, COMSIG_MOVABLE_MOVED)
+
+/datum/component/proximity_monitor/PostTransfer()
+	if(!ismovable(parent) && !isturf(parent))
+		return COMPONENT_INCOMPATIBLE
+	host = null
+	hasprox_receiver = parent
+	if(!isturf(hasprox_receiver.loc))
+		host = get_atom_on_turf(host)
+	recenter_prox_checkers()
 
 /**
- * Called when the `parent` receives the `COMSIG_MOVABLE_MOVED` signal, which occurs when it `Move()`s
- *
- * Code is only ran when there is no `Dir`, which occurs when the parent is teleported, gets placed into a storage item, dropped, or picked up.
- * Normal movement, for example moving 1 tile to the west, is handled by the `proximity_checker` objects.
+ * Called when the `hasprox_receiver` moves. If a `host` exists, its movement will also call this proc.
  *
  * Arguments:
- * * source - this will be the `parent`
- * * OldLoc - the location the parent just moved from
- * * Dir - the direction the parent just moved in
+ * * datum/source - this will be the `parent`
+ * * atom/old_loc - the location the parent just moved from
+ * * dir - the direction the parent just moved in
  * * forced - if we were forced to move
  */
-/datum/component/proximity_monitor/proc/HandleMove(datum/source, atom/OldLoc, Dir, forced)
-	if(!Dir) // No dir means the parent teleported, or moved in a non-standard way like getting placed into disposals, onto a table, dropped, picked up, etc.
-		recenter_prox_checkers()
+/datum/component/proximity_monitor/proc/handle_move(datum/source, atom/old_loc, dir, forced)
+	SIGNAL_HANDLER
+
+	if(dir)
+		move_prox_checkers(dir)
+		return // It was just a normal tile-based move, return.
+
+	// The field is always active while the receiver is on a turf.
+	if(isturf(hasprox_receiver.loc))
+		toggle_checkers(TRUE)
+		if(host && ismovable(host))
+			UnregisterSignal(host, COMSIG_MOVABLE_MOVED)
+			host = null
+	// The receiver moved inside of another atom, set the host to that atom so we can track its movement.
+	else if(always_active)
+		toggle_checkers(TRUE)
+		var/atom/new_host = get_atom_on_turf(hasprox_receiver)
+		if(ismovable(new_host) && host != new_host && host != old_loc)
+			host = new_host
```

Comment:
You'd have to unregister from the host as well here right?
Prob best to make unsetting the host a proc

Reply:
I'm actually reworking most of this section to make `always_active` components not horribly break when dealing with nested storage, so this will change quite a bit. Good idea about making the host setting a proc though.